### DISCUSSION TITLE
sudo command fails on client

### DISCUSSION THREAD

**Discussion by brcisna** [ltsp/ltsp#337] [Q&A]
Hello All,
Debian Bullseye

After doing numerous updates over the last sveral days rebuilt client image.
I can log into the clients as any user as expected.
The users that i have setup for sudo can no longer run sudo command on client. It does work as usual on the sevrer though. Have never had to troubleshoot this before.
Example:  when i run 'sudo kill -9 ### ' ,, then password entered,,  i   get 'Login incorrect'
I did double check to make sure the sudousers file hasnt been relaced by updates,,,which it has in the pst,,and it is still the same for a few months now.

Also, I did rebuild the client image a second time just to try something, no changes.

Thank You

**Comment by alkisg** [comment]
Can they run `su - user`, to authenticate as themselves?
Also, run `journalctl -f` as root on the client (e.g. via epoptes), then try to run `sudo` as the user, and check what the journal logs have to say about it.

**Comment by brcisna** [comment]
Hi Alkisg,

Thank You for helping out.

When i run the first command on client I get the following :
su - sudousername
Password: 
.Login incorrect.
su: Authentication failure

When I run the second pair of commands you suggested in the journalctl logs i see one line generated,this line:
Nov 26 06:37:12 hostname sshd[243636]: Connection closed by authenticating user sudousername #.#.#.# port 57862 [preauth]

Also adding I know the sudoers file is correct on the server were the two sudousers function as expected,,but I cannot actually see the sudoers file is identical as i cannot get root on client but the dates are identical on each machine so i assume they are both identical. I havent touched this file since server build a little over year ago,other than when it was replaced a few months ago by  by updates and edited at that time since that time has remained the same so im sure it isnt fatfingered in that regard.

Have not changed any firewall rules for several months for a little more info. Have added many many multimedia programs each month as i use this server/client for doing novice video editing for history videos. Cannot picture any of these apps causing sudo problems under the hood.

Googled and dont really see this exact error 'login incorrect' for trying to run sudi command. This just started two days ago. FYI.

Ideas?

EDIT:
Alkisg, Please look at screenshot. attached .  I did see something interesting in /etc/ssh in that there is another sshd_config.ucf.dist added just 7 days ago, Don't think that could be affecting anything for sudoers but dont know what this would have been added from other than updates? 

Thanks again

![sshd_config ucf-dist](https://user-images.githubusercontent.com/47399290/100356137-70ecff00-2fb8-11eb-93c7-f6f0fed8bcb6.png)

**Comment by alkisg** [comment]
https://github.com/ltsp/ltsp/issues are for LTSP bugs,
so I converted this question into a discussion: https://github.com/ltsp/ltsp/discussions.

Your issue isn't related at all with sudoers. It's about authentication. `su - user` just authenticates, it's like `login`, the user doesn't need to be a sudoer to do that.

Authentication in LTSP is done by SSH'ing to the LTSP server. So, try to SSH to the LTSP server with these credentials from another box. Does it work? If your SSH server is broken, then LTSP client authentication will break as well.

That ucf-dist file means that your sshd config was modified when it was upgraded.

**Comment by brcisna** [comment]
Hi Alkis,

Yes, From the client i can ssh fine to the server via ssh with the same two sudo users credentials, That is the puzzler. It is strictly on the client. As i said for example sometimes,,not very often VLC will hang at the taskbar when looking at a lot of videos for editing ,, so i have to run kill -9 (process) as sudo .This is now,, where i get the '.Incorrent Login. only on client machine. Sudo works fine on server.

For completeness. the same sudo users are the ones that log into client without issue as well. BUT the sudo command get the authentication error "Login Incorrect" 

Thank You . I will try and put my future questions here in discussions.

**Comment by brcisna** [comment]
Hello All,

Just wanted to mention once again this problem still exists on the ltsp image/ client machine only after many updates and image rebuilds.
Probably not neccesarily to do with LTSP itself.. but the same two ,,users that are sudoers can function on the server but not the client. Not sure what the malfunction is.
The two users that cannot sudo on client can log into any client machine fine.

Thanks

**Comment by alkisg** [comment]
You can come to IRC at some point if you'd like me to troubleshoot this over VNC.
https://ltsp.org/ => click on IRC.

**Comment by alkisg** [comment]
P.S. @brcisna note that I'm in Greek timezone (UTC+2); I saw you came to IRC around 02:30 AM my time, so I was sleeping! :)

**Comment by martadinata666** [comment]
Hello, I also got this issue. After further investigation, ltsp client fail login and sudo command fail 'login incorrect.' happen on openssh 8.4 (debian sid). Then I manually compile openssh 8.3, the login and sudo work no issue. May you please help me confirm this?
Thanks.

**Comment by alkisg** [comment]
@martadinata666, brcisna's issue was that he had some process running as "user" **before** the user had logged in. This made the /home/user directory be in use, so pamltsp couldn't mount /home/user from the server via SSHFS.

I haven't tested Debian sid recently. You too may come to IRC if you would like remote support via epoptes reverse connection.

**Reply by brcisna** [reply]
Hello,

I wanted to add in here, for just some details to add to this scenario. Both sudo  users that fails with sudo commands (on client) can login to the client no problem. However, if i purposely enter a wrong password for either of the users,at login,one time ,then enter the correct password the password is never accepted .The client has to be rebooted. , I have tried this multiple times to confirm so this can be easily reproduced. 
Not too sure an underlyinh problem may be python versions updates as also Catfish fails to open with strange python errors,,but i know this has nothing to do with sshfs as such.Just trying to ad a few details. Also on client machine nothing is ever logged to syslog & auth.log ,although there may be something in one of th ltsp scripts that sends client logs to server logs,cant remember if this is the case. 

Thank You
